hudi-spark/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -93,6 +93,11 @@ private[hudi] object HoodieSparkSqlWriter {
       operation = WriteOperationType.INSERT
     }
 
+    // If the mode is Overwrite, should use INSERT_OVERWRITE operation
```

Comment:
I think this won't work. Insert overwrite only updates partitions that have records in the dataframe. Other partitions continue to have old data.

But, I like the idea. Maybe we can add additional configuration to insert_overwrite to mark old partitions as 'deleted'? This can be done in a way to support https://issues.apache.org/jira/browse/HUDI-1350.

Reply:
@satishkotha  thanks, 
1. i think the method is ok . Because in HoodieSparkSqlWriter.scala , if we set  operation as replace action, then it also use  "client.insertOverwrite(hoodieRecords, instantTime); "  here will do as " Insert overwrite only updates partitions".

 public static HoodieWriteResult doWriteOperation(SparkRDDWriteClient client, JavaRDD<HoodieRecord> hoodieRecords,
                                                   String instantTime, WriteOperationType operation) throws HoodieException {
    switch (operation) {
      case BULK_INSERT:
        Option<BulkInsertPartitioner> userDefinedBulkInsertPartitioner =
                createUserDefinedBulkInsertPartitioner(client.getConfig());
        return new HoodieWriteResult(client.bulkInsert(hoodieRecords, instantTime, userDefinedBulkInsertPartitioner));
      case INSERT:
        return new HoodieWriteResult(client.insert(hoodieRecords, instantTime));
      case UPSERT:
        return new HoodieWriteResult(client.upsert(hoodieRecords, instantTime));
      case INSERT_OVERWRITE:
        return client.insertOverwrite(hoodieRecords, instantTime);
      default:

2. https://issues.apache.org/jira/browse/HUDI-1350. is another job.

Reply:
@lw309637554  can you add below test:

step1: Write N records to hoodie table for partition1
step2: Write N more records using SaveMode.Overwrite for partition2
step3: Query for all the rows from hoodie table. We should only see N records for partition2.  

If i understand your code correctly, with this change, we will see 2N records

Reply:
@satishkotha thanks, have add the unit test 
and it is okay